fv3core/stencils/remapping.py

```diff
@@ -1,128 +1,521 @@
-    )
-
-    remapping_part_2 = utils.cached_stencil_class(VerticalRemapping2)(
-        pfull=pfull, cache_key="remapping_part2"
-    )
-
-    remapping_part_1(
-        tracers,
-        pt,
-        delp,
-        delz,
-        peln,
-        u,
-        v,
-        w,
-        ua,
-        cappa,
-        q_con,
-        pkz,
-        pk,
-        pe,
-        hs,
-        dp1,
-        ps,
-        wsd,
-        omga,
-        ak,
-        bk,
-        gz,
-        cvm,
-        ptop,
-    )
-    remapping_part_2(
-        tracers["qvapor"],
-        tracers["qliquid"],
-        tracers["qice"],
-        tracers["qrain"],
-        tracers["qsnow"],
-        tracers["qgraupel"],
-        q_cld,
-        pt,
-        delp,
-        delz,
-        peln,
-        u,
-        v,
-        w,
-        ua,
-        cappa,
-        q_con,
-        gz,
-        pkz,
-        pk,
-        pe,
-        hs,
-        te0_2d,
-        dp1,
-        cvm,
-        pfull,
-        ptop,
-        akap,
-        zvir,
-        last_step,
-        bdt,
-        mdt,
-        consv_te,
-        do_adiabatic_init,
-    )
+    with computation(PARALLEL), interval(...):
+        delp = dp2
+        pn2 = log(pe2)
+        pk = exp(akap * pn2)
+
+
+def pressures_mapu(
+    pe: FloatField,
+    pe1: FloatField,
+    ak: FloatFieldK,
+    bk: FloatFieldK,
+    pe0: FloatField,
+    pe3: FloatField,
+):
+    with computation(BACKWARD):
+        with interval(-1, None):
+            pe_bottom = pe
+            pe1_bottom = pe
+        with interval(0, -1):
+            pe_bottom = pe_bottom[0, 0, 1]
+            pe1_bottom = pe1_bottom[0, 0, 1]
+    with computation(FORWARD):
+        with interval(0, 1):
+            pe0 = pe
+        with interval(1, None):
+            pe0 = 0.5 * (pe[0, -1, 0] + pe1)
+    with computation(FORWARD), interval(...):
+        bkh = 0.5 * bk
+        pe3 = ak + bkh * (pe_bottom[0, -1, 0] + pe1_bottom)
+
+
+def pressures_mapv(
+    pe: FloatField, ak: FloatFieldK, bk: FloatFieldK, pe0: FloatField, pe3: FloatField
+):
+    with computation(BACKWARD):
+        with interval(-1, None):
+            pe_bottom = pe
+        with interval(0, -1):
+            pe_bottom = pe_bottom[0, 0, 1]
+    with computation(FORWARD):
+        with interval(0, 1):
+            pe3 = ak
+            pe0 = pe
+        with interval(1, None):
+            bkh = 0.5 * bk
+            pe0 = 0.5 * (pe[-1, 0, 0] + pe)
+            pe3 = ak + bkh * (pe_bottom[-1, 0, 0] + pe_bottom)
+
+
+def update_ua(pe2: FloatField, ua: FloatField):
+    with computation(PARALLEL), interval(0, -1):
+        ua = pe2[0, 0, 1]
+
+
+def copy_from_below(a: FloatField, b: FloatField):
+    with computation(PARALLEL), interval(1, None):
+        b = a[0, 0, -1]
+
+
+def sum_te(te: FloatField, te0_2d: FloatField):
+    with computation(FORWARD):
+        with interval(0, None):
+            te0_2d = te0_2d[0, 0, -1] + te
+
+
+class VerticalRemapping:
```

Comment:
It would be good to keep the fortran reference here. I believe this should be lagrangian_to_eulerian in fv_mapz.F90

Reply:
Done!